### DISCUSSION TITLE
tutorials/3d/csg_tools

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#8] [User-contributed notes]
# tutorials/3d/csg_tools

CSG stands for Constructive Solid Geometry, and is a tool to combine basic shapes or custom meshes to create more complex shapes. In 3D modeling software, CSG is mostly known as "Boolean Operators"...

https://docs.godotengine.org/en/latest/tutorials/3d/csg_tools.html

**Comment by 31** [comment]
[Moved to [Document the CSG system's expected level of reliability #8767](https://github.com/godotengine/godot-docs/issues/8767) after godot-docs-user-notes rule change.]

**Comment by ghost** [comment]
Video about process of exporting CSG to Blender and back to Godot https://www.youtube.com/watch?v=gIK02BgCCBI

**Comment by alxrbrs** [comment]
In regards to the prototyping exercise;  make sure to drag all of the room objects(bed, desk, lamp, bookshelf) out from the "room" parent node so that they're displayed once you're done.

**Comment by erwan-privat** [comment]
> To convert a CSG node to a MeshInstance3D node, select it, then choose CSG > Bake Mesh Instance in the toolbar.

I'm using 4.4.1.stable and it might be right under my eyes for the life of me I cannot find the bake option. Anyone has a screenshot? Thanks.

**Reply by Calinou** [reply]
Yes, here it is on 4.4.1.stable:

![image](https://github.com/user-attachments/assets/270dd1b2-8b2d-4d6a-a3db-2c043cee0bef)

Remember that the option is only visible on the root CSG node in a CSG tree. In the example, you'll see the option on CSGBox3D but not CSGSphere3D because of this.

**Comment by Devilhanilhan** [comment]
I'm you glad you added  method baking csg to MeshInstance3d. So it will be cool if you add automatically Manifolding feature like blender. 

I want note CSG worked without manifolding before, on preview versions of godot. Because of adding new features you required Manifold. But it will be cool if this worked without requiring manifold or adding with automatically Manifolding button with him

**Reply by Calinou** [reply]
Feature proposals should be made on the [Godot proposals repository](https://github.com/godotengine/godot-proposals), not here.
